message_test.go

```diff
@@ -1415,21 +1415,17 @@ func Test_getFieldIndex(t *testing.T) {
 			F1 string
 		}{}).Elem()
 
-		index, err := getFieldIndex(st.Type().Field(0))
-
-		require.NoError(t, err)
-		require.Equal(t, 1, index)
+		indexTag := field.NewIndexTag(st.Type().Field(0))
+		require.Equal(t, 1, indexTag.Id)
 	})
 
 	t.Run("returns index from field tag instead of field name when both match", func(t *testing.T) {
```

Comment:
as we now have `field/index_tag.go` I think we should move all related tests into `field/index_tag_test.go` and remove them from other places. In index tag test we can test different test cases as well using table tests.